doc/source/dev/api-dev/array_api.rst

```diff
@@ -217,11 +217,12 @@ The following pytest markers are available:
 
 * ``array_api_compatible -> xp``: use a parametrisation to run a test on
   multiple array backends.
-* ``skip_if_array_api``: don't run a test if ``SCIPY_ARRAY_API`` is on.
+* ``skip_if_xp_not_numpy(reason=None)``: only run a test with NumPy backend.
```

Comment:
I should clarify that this means only run with `numpy` i.e. not `numpy.array_api`. Unless someone starts writing unconventional tests with strange `xp` values, the `xp` values come from the `array_api_compatible` decorator which will give a subset of `{np, numpy.array_api, torch, cupy}`.

The point of the decorator is to indicate that the test only checks for compatibility with regular `numpy` arrays, not any other backend that we are starting to support experimentally. For some of the functions in `cluster` which still only support `numpy` for whatever reason, we should use this instead of skipping the test entirely.